src/vcall.cpp

```diff
@@ -288,12 +289,12 @@ uint32_t jitc_var_vcall(const char *name, uint32_t self, uint32_t mask,
         for (uint32_t j = 0; j < n_out; ++j)
             jitc_var_vcall_collect_data(vcall->data_map, data_size, i,
                                         out_nested[j + i * n_out],
-                                        vcall->use_self);
```

Comment:
Shouldn't we have a logical or on the `optix` value here ?
In case we already have it set to `true`, but the `jitc_var_vcall_collect_data` writes `false`.